command/kv_metadata_patch.go

```diff
@@ -0,0 +1,193 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+var (
+	_ cli.Command             = (*KVMetadataPutCommand)(nil)
+	_ cli.CommandAutocomplete = (*KVMetadataPutCommand)(nil)
+)
+
+type KVMetadataPatchCommand struct {
+	*BaseCommand
+
+	flagMaxVersions        int
+	flagCASRequired        BoolPtr
+	flagDeleteVersionAfter time.Duration
+	flagCustomMetadata     map[string]string
+	testStdin              io.Reader // for tests
+}
+
+func (c *KVMetadataPatchCommand) Synopsis() string {
+	return "Patches key settings in the KV store"
+}
+
+func (c *KVMetadataPatchCommand) Help() string {
+	helpText := `
+Usage: vault metadata kv patch [options] KEY
+
+  This command can be used to create a blank key in the key-value store or to
+  update key configuration for a specified key.
+
+  Create a key in the key-value store with no data:
+
+      $ vault kv metadata patch secret/foo
+
+  Set a max versions setting on the key:
+
+      $ vault kv metadata patch -max-versions=5 secret/foo
+
+  Set delete-version-after on the key:
+
+      $ vault kv metadata patch -delete-version-after=3h25m19s secret/foo
+
+  Require Check-and-Set for this key:
+
+      $ vault kv metadata patch -cas-required secret/foo
+
+  Set custom metadata on the key:
+
+      $ vault kv metadata patch -custom-metadata=foo=abc -custom-metadata=bar=123 secret/foo
+
+  Additional flags and more advanced use cases are detailed below.
+
+` + c.Flags().Help()
+	return strings.TrimSpace(helpText)
+}
+
+func (c *KVMetadataPatchCommand) Flags() *FlagSets {
+	set := c.flagSet(FlagSetHTTP | FlagSetOutputFormat)
+
+	// Common Options
+	f := set.NewFlagSet("Common Options")
+
+	f.IntVar(&IntVar{
+		Name:    "max-versions",
+		Target:  &c.flagMaxVersions,
+		Default: -1,
```

Comment:
Why are the defaults here -1 instead of 0?

Reply:
We do not want to overwrite the field if the user did not provide the flag. Providing 0 is a valid value for the `max-versions` flag. There is a check further on in the command handler that will check for >= 0 to determine whether or not to provide the field in the request body.